cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -98,15 +98,29 @@ func (c *CStorPoolController) syncHandler(key string, operation common.QueueOper
 
 // cStorPoolEventHandler is to handle cstor pool related events.
 func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperation, cStorPoolGot *apis.CStorPool) (string, error) {
+	var zpoolDumpErr error
 	pool.RunnerVar = util.RealRunner{}
+	uid := string(cStorPoolGot.GetUID())
+	csp := pool.CStorPools[uid]
 	switch operation {
 	case common.QOpAdd:
 		glog.Infof("Processing cStorPool added event: %v, %v", cStorPoolGot.ObjectMeta.Name, string(cStorPoolGot.GetUID()))
 
 		// lock is to synchronize pool and volumereplica. Until certain pool related
 		// operations are over, the volumereplica threads will be held.
 		common.SyncResources.Mux.Lock()
+		if csp != nil {
+			c.recorder.Event(cStorPoolGot, corev1.EventTypeWarning, string(common.AlreadyPresent), string(common.MessageResourceAlreadyPresent))
+		}
 		status, err := c.cStorPoolAddEventHandler(cStorPoolGot)
+		if status == string(apis.CStorPoolStatusOnline) {
+			pool.CStorPools[uid] = cStorPoolGot.DeepCopy()
+			pool.CStorZpools[uid], zpoolDumpErr = pool.ZpoolDump()
+			if zpoolDumpErr != nil {
+				glog.Errorf("failed in getting zpool dump %v", zpoolDumpErr)
```

Comment:
Should this be a `Info` as we are handling this anyway?
If this is critical to functioning then `Error` is fine.

Reply:
this being error path, needed for debugging

Reply:
this command failing means that something is wrong in the system